### ISSUE TITLE
Infinite loop into Component.js

### ISSUE THREAD

**Issue by westtrade** [marko-js/marko#749]
https://github.com/marko-js/marko/blob/master/src/components/Component.js#L553

If Component don't have id that place become an infinite loop.

**Comment by austinkelleher** [comment]
I'm actually looking into this exact issue right now. Related:

https://github.com/marko-js/marko/issues/690

**Comment by austinkelleher** [comment]
Are you using a style tag at the root?

**Comment by westtrade** [comment]
No, i don't use anyone style tags into components, but have this error too.

**Comment by austinkelleher** [comment]
Can you please post a small example of a component that is causing you this issue? Thanks.

**Comment by westtrade** [comment]
Yep,
Error has happening if i create a component without class
```
<div class="layout">
	<div if(input.sidebar.renderBody) class="sidebar">
		<include(input.sidebar)/>
	</div>
	<main if(input.content.renderBody) class="content">
		<include(input.content)/>
	</main>
</div>
```
- in that case - i have error

```

class {
	
}

<div class="layout">
	<div if(input.sidebar.renderBody) class="sidebar">
		<include(input.sidebar)/>
	</div>
	<main if(input.content.renderBody) class="content">
		<include(input.content)/>
	</main>
</div>

``` - works fine

**Comment by westtrade** [comment]
I think real bug somewhere in generator functions. But - as i think - was better if in this place, error was thrown if id does not exists than danger of infinite loop

**Comment by austinkelleher** [comment]
Fixed in `marko@4.4.18`
